cohortextractor/emis_backend.py

```diff
@@ -744,73 +744,51 @@ def _none_of_these_codes_occur_on_same_day(self, joined_table, codelist):
         """
 
     def patients_registered_practice_as_of(self, date, returning=None):
+        # At the moment we can only return current values for the fields in question.
+
         if returning == "stp_code":
-            column = "STPCode"
+            column = "stp_code"
         elif returning == "msoa_code":
-            column = "MSOACode"
-        elif returning == "nhse_region_name":
-            column = "Region"
-        elif returning == "pseudo_id":
-            column = "Organisation_ID"
+            column = "msoa"
+        elif returning == "nuts1_region_name":
+            column = "english_region_name"
         else:
             raise ValueError(f"Unsupported `returning` value: {returning}")
-        # Note that current registrations are recorded with an EndDate of
-        # 9999-12-31. Where registration periods overlap we use the one with
-        # the most recent start date. If there are several with the same start
-        # date we use the longest one (i.e. with the latest end date).
+
         return (
             ["patient_id", returning],
             f"""
             SELECT
-              "registration_id" AS patient_id,
-              Organisation.{column} AS {returning}
-            FROM (
-              SELECT "registration_id", Organisation_ID,
-              ROW_NUMBER() OVER (
-                PARTITION BY "registration_id" ORDER BY StartDate DESC, EndDate DESC
-              ) AS rownum
-              FROM RegistrationHistory
-              WHERE StartDate <= {quote(date)} AND EndDate > {quote(date)}
-            ) t
-            LEFT JOIN Organisation
-            ON Organisation.Organisation_ID = t.Organisation_ID
-            WHERE t.rownum = 1
+              id AS patient_id,
+              {column} AS {returning}
+            FROM
+              patient
             """,
         )
 
     def patients_address_as_of(self, date, returning=None, round_to_nearest=None):
-        # N.B. A value of -1 indicates no postcode recorded on the
-        # record, an invalid postcode, or no fixed abode.
-        #
-        # Related, there is a column in the address table to indicate
-        # NP for no postcode or NFA for no fixed abode
+        # At the moment we can only return current values for the fields in question.
+
+        if date != "today":
```

Comment:
I was surprised this worked at all because since https://github.com/opensafely/cohort-extractor/pull/255/commits/bd9c7be9ab440a97d86803e7fc4d87805bb9bef9 date expressions are all supposed to be transformed into ISO strings by the time they reach the backend. What I've realised is that this isn't working for `address_as_of` because it uses the argument `date` rather than `reference_date` which is a bug that I need to fix. But when that's fixed this will stop working.

The simplest change to retain roughly the same behaviour would be to calll `datetime.date.today()` here and compare the result. Maybe we could relax the condition slightly and just require that the date is within a certain distance (one month?) of today.

Reply:
Good idea, I've fixed this -- could you re-review when you get the chance?